integration-tests/pipelines/konflux-e2e-tests.yaml

```diff
@@ -0,0 +1,249 @@
+      default: ''
+    - name: ocp-version
+      description: 'The OpenShift version to use for the ephemeral cluster deployment.'
+      type: string
+    - name: test-event-type
+      description: 'Indicates if the test is triggered by a Pull Request or Push event.'
+      default: 'none'    
+    - name: konflux-test-infra-secret
+      description: The name of secret where testing infrastructures credentials are stored.
+      type: string
+    - name: cloud-credential-key
+      type: string
+      description: The key secret from konflux-test-infra-secret where all AWS ROSA configurations are stored.
+    - name: replicas
+      description: 'The number of replicas for the cluster nodes.'
+      type: string
+    - name: machine-type
+      description: 'The type of machine to use for the cluster nodes.'
+      type: string
+    - name: oci-container-repo
+      default: 'quay.io/konflux-test-storage/konflux-team/integration-service'
+      description: The OCI container used to store all test artifacts.
+    - name: quality-dashboard-api
+      default: 'none'
+      description: 'Contains the url of the backend to send metrics for quality purposes.'
+    - name: component-image
+      default: 'none'
+      description: 'Container image built from any konflux git repo. Use this param only when you run Konflux e2e tests
+        in another Konflux component repo. Will pass the component built image from the snapshot.'
+  tasks:
+    - name: rosa-hcp-metadata
+      taskRef:
+        resolver: git
+        params:
+          - name: url
+            value: https://github.com/konflux-ci/konflux-qe-definitions.git
+          - name: revision
+            value: main
+          - name: pathInRepo
+            value: common/tasks/rosa/hosted-cp/rosa-hcp-metadata/rosa-hcp-metadata.yaml
+    - name: test-metadata
+      taskRef:
+        resolver: git
+        params:
+          - name: url
+            value: https://github.com/konflux-ci/konflux-qe-definitions.git
+          - name: revision
+            value: main
+          - name: pathInRepo
+            value: common/tasks/test-metadata/0.1/test-metadata.yaml
+      params:
+        - name: SNAPSHOT
+          value: $(params.SNAPSHOT)
+        - name: test-name
+          value: $(context.pipelineRun.name)
+    - name: create-oci-container
+      taskRef:
+        resolver: git
+        params:
+          - name: url
+            value: https://github.com/konflux-ci/konflux-qe-definitions.git
+          - name: revision
+            value: main
+          - name: pathInRepo
+            value: common/tasks/create-oci-artifact/0.1/create-oci-artifact.yaml
+      params:
+        - name: oci-container-repo
+          value: $(params.oci-container-repo)
+        - name: oci-container-tag
+          value: $(context.pipelineRun.name)
+    - name: provision-rosa
+      when:
+        - input: $(tasks.test-metadata.results.test-event-type)
```

Comment:
Just as an info: it's now possible to also set the `contexts` field for IntegrationTestScenarios` which allows you to set it to only execute in case of a Pull requests - see more in the [docs](https://konflux-ci.dev/docs/how-tos/testing/integration/choosing-contexts/).

Reply:
We are planning to move to contexts as next step. I will open a separated Pull Request